CIP-?/README.md

```diff
@@ -0,0 +1,109 @@
+---
+CIP: ?
+Title: Cardano URI Scheme Extensions
+Status: Proposed
+Category: Tools
+Authors:
+    - Pi Lanningham <pi@sundaeswap.finance>
+Implementors: N/A
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/556
+Created: 2020-09-22
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This extends CIP-13, which describes a Cardano URI scheme, with several more objects that would be useful to be able to assign addresses to, in particular blocks, transactions, transaction metadata, and transaction outputs.
+
+## Motivation: why is this CIP necessary?
+
+CIP-13 defined two initial URL authorities, for payment links and delegating to a stakepool.
+
+However, in a number of contexts, it would be useful to canonically link to other Cardano objects, such as:
+- Providing links to a transaction, to be opened in a wallet or chain explorer of the users choice
+- To provide richly interconnected metadata, such as in [CIP-100](https://github.com/cardano-foundation/CIPs/pull/556)
+
+Without a canonical standard for how to define these URIs, these objects are either unaddressable, not machine-interpretable, or will suffer from a divergence of convention. Similarly, we seek to fit an existing structure, such as CIP-0013, to reduce the number of different conventions that need be supported by the ecosystem.
+
+## Specification
+
+We extend CIP-13 with 2 new authorities for referencing blocks and transactions.
+
+Examples:
+```
+web+cardano://block/c6a8976125193dfae11551c5e80a217403d953c08ebbd69bba904d990854011f
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9#1
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9/metadata
+web+cardano://transaction/d3616b772c91f118346e74863d1722810a4858e4d7cc7663dc2eed345d7bca72/metadata/674
+web+cardano://transaction/self/metadata/1694
+```
+
+### Grammar & interpretation
+
+We extend the grammar from CIP-0013 with two new authorities:
+
+```
+authorityref = (stakepoolref | blockref | transactionref | otherref)
+```
+
```

Comment:
```suggestion
```
This isn't needed because you are only defining grammars for the cases you are covering here: it must be one or the other, and so the `stakepoolref` and `otherref` cases are irrelevant and the definitions for `blockref` and `transactionref` below are sufficient.

Also: for all CIPs extending the CIP-0013 URI Scheme to be equally meaningful without ordering, I wouldn't want further CIPs besides this one to try to list all the previously defined authorities (note we also already have https://github.com/cardano-foundation/CIPs/pull/546 in the works, which defines `claimtokenref` and should likewise be considered irrelevant).

Reply:
I pictured this more as a delta, to show where it fits within the existing BNF; BNF doesn't really have a good mechanism for this, afaik.

Technically speaking, CIP-0013 is written in such a way that this doesn't need a grammar at all; `otherref` is defined super generally, and so every one of the url patterns I defined would be covered by "otherref", I think.  So, to be clear and precise, `authorityref` needs to be updated, otherwise my BNF would just be dangling and unreferenced.

That being said, it's more for human consumption, and not being consumed programatically, so I'm happy to conform to whatever convention you see fit. To prevent a bunch of back-and-forth, feel free to just update it to the style you prefer.  Perhaps something like `authorityref = (blockref | transactionref | ...)` to indicate "and all the rest from other CIPs"?

Reply:
yes, I know what you mean; it's hard to put into words how ABNF isn't written with "hooks" or "plugins" in mind.  I'd tried to rewrite CIP-0013 that way and failed.  It was my hope that no provider of further CIPs would have to include any BNF line that was in CIP-0013 and I do think that it all fits together with the redundant `authorityref` definition above removed.

If that seems incomplete or too abrupt to you, even if `authorityref = (blockref | transactionref | ...)` is not strictly valid ABNF then I think it would work well here... as would a verbal description like "This CIP adds two new possibilities for the `authorityref` term." 🤔

Reply:
Given that we don't really expect to consume the ABNF programatically, I'm personally in favor of that syntax, as I think it's immediately clear what it means.

Reply:
I think I would be in favor of:
`authorityref = (... | blockref | transactionref)`

From a purely programmatic standpoint to me this means "everything that came before plus this"

I would agree with Pi's assessment that since this is merely syntactic sugar for human readers it's not crucially important either way and more important that we get the meaning across more than anything.